gnomad_qc/v3/sample_qc/subpop_analysis.py

```diff
@@ -0,0 +1,327 @@
+    :param min_af: Minimum population variant allele frequency to retain variant in QC MT
+    :param min_inbreeding_coeff_threshold: Minimum site inbreeding coefficient to retain variant in QC MT
+    :param min_hardy_weinberg_threshold: Minimum site HW test p-value to keep
+    :param ld_r2: Minimum r2 to keep when LD-pruning (set to `None` for no LD pruning)
+    :return: Filtered QC MT with sample metadata for the specified population to use for subpop analysis
+    """
+    meta_ht = meta.ht()
+
+    # Add info to the MT
+    info_ht = get_info(split=False).ht()
+    info_ht = info_ht.annotate(
+        info=info_ht.info.select(
+            # No need for AS_annotations since it's bi-allelic sites only
+            **{x: info_ht.info[x] for x in info_ht.info if not x.startswith("AS_")}
+        )
+    )
+    mt = mt.annotate_rows(info=info_ht[mt.row_key].info)
+
+    # Add sample metadata to the QC MT
+    mt = mt.annotate_cols(**meta_ht[mt.col_key])
+
+    # Filter the  QC MT to the specified pop
+    pop_mt = mt.filter_cols(mt.population_inference.pop == pop)
+
+    # Remove hard filtered samples
+    pop_mt = pop_mt.filter_cols(~pop_mt.sample_filters.hard_filtered)
+    pop_mt = pop_mt.filter_rows(hl.agg.any(pop_mt.GT.is_non_ref()))
+
+    # Generate a QC MT for the given pop
+    pop_qc_mt = get_qc_mt(
+        pop_mt,
+        min_af=min_af,
+        min_inbreeding_coeff_threshold=min_inbreeding_coeff_threshold,
+        min_hardy_weinberg_threshold=min_hardy_weinberg_threshold,
+        ld_r2=ld_r2,
+    )
+
+    return pop_qc_mt
+
+
+def annotate_subpop_meta(ht: hl.Table) -> hl.Table:
+    meta_ht = meta.ht()
+    ht = ht.annotate(**meta_ht[ht.key])
+    ht = ht.annotate(
+        subpop_description=ht.project_meta.subpop_description,
+        v2_pop=ht.project_meta.v2_pop,
+        v2_subpop=ht.project_meta.v2_subpop,
+        pop=ht.population_inference.pop,
+        project_id=ht.project_meta.project_id,
+        project_pop=ht.project_meta.project_pop,
+    )
+
+    return ht
+
+
+def main(args):  # noqa: D103
+    pop = args.pop
+    include_unreleasable_samples = args.include_unreleasable_samples
+    high_quality = args.high_quality
+    # Read in the raw mt
+    mt = get_gnomad_v3_mt(key_by_locus_and_alleles=True)
+
+    # Filter to test partitions if specified
+    if args.test:
+        logger.info("Filtering to the first two partitions of the MT")
+        mt = mt._filter_partitions(range(2))
+
+    # Write out the densified MT
+    if args.make_full_subpop_qc_mt:
+        logger.info("Generating densified MT to use for all subpop analyses...")
+        mt = compute_subpop_qc_mt(mt, args.min_popmax_af)
+        mt = mt.write(
+            get_checkpoint_path("test_make_full_subpop_qc", mt=True)
+            if args.test
+            else subpop_qc.path,
+            overwrite=args.overwrite,
+        )
+
+    if args.run_subpop_pca:
+        logger.info("Filtering subpop QC MT...")
+        mt = subpop_qc.mt()
```

Comment:
Only way to have this be a test is if we use the test MT
```suggestion
        if args.test:
            mt = hl.read_matrix_table(get_checkpoint_path("test_make_full_subpop_qc", mt=True))
        else:
            mt = subpop_qc.mt()
```